components/Boards/PlayerVsPlayer.js

```diff
@@ -23,53 +28,49 @@ export default function RandomMoveEngine() {
     setGame(new Chess());
   }, []);
 
-  // // ---CREATE SEPERATE GAME HISTORY TO BYPASS GAME RESETS---
-  function historyStorage() {
-    if (
-      latestHistory?.color !==
-      game?.history({ verbose: true })[
-        game.history({ verbose: true }).length - 1
-      ]?.color
-    ) {
-      setHistory([
-        ...history,
-        game?.history({ verbose: true })[
-          game.history({ verbose: true }).length - 1
-        ],
-      ]);
+  // // ---CREATE SEPERATE MOVE/FEN HISTORY TO BYPASS GAME RESETS---
+  function historyStorage(latestResult) {
+    if (latestResult !== null) {
+      setMoveHistory([...moveHistory, latestResult]);
     }
   }
-
-  // // ---CHECK FOR ZOMBIE PIECE => RESPAWN ZOMBIE => RESET GAME OBJECT WITH .fen()---
-  //  ---use previousMove to check captures (flags = "c")
+  //Without useEffect,fenHistory only updates one before last
   useEffect(() => {
-    if (previousMove?.flags === "c") {
-      game.put(
-        { type: previousMove.captured, color: previousMove.color },
-        previousMove.from
-      );
-      const newGame = new Chess(game.fen());
-      setGame(newGame);
-      setFen(newGame.fen());
-      historyStorage();
+    setFenHistory([...fenHistory, fen]);
+  }, [fen]);
+
+  // ---ZOMBIE FUNCTION => RESPAWN ZOMBIE => RESET GAME OBJECT WITH .fen()---
+  function zombieMove(latestResult, game) {
+    game.put(
+      { type: latestResult.captured, color: latestResult.color },
+      latestResult.from
+    );
+    const newGame = new Chess(game.fen());
+    setGame(newGame);
+    setFen(newGame.fen());
+
+    const possibleMoves = game?.moves();
+    if (game.game_over() || game.in_draw() || possibleMoves.length === 0) {
+      return; // exit if the game is over
     }
-  }, [previousMove]);
+  }
 
   // // ---MAKE A MOVE AND UPDATE GAME OBJECT---
   function makeAMove(move) {
     const gameCopy = { ...game };
     const result = gameCopy.move(move);
-    // setGame(gameCopy);
+    // setGame(gameCopy); //This one seems to be optional?
```

Comment:
I appreciate a lot of debugging went into this, but do we need this comment in `main`?